example-widget-mui/src/UploadImagePage/UploadImagePage.tsx

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright 2024 Nordeck IT + Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { MuiCapabilitiesGuard } from '@matrix-widget-toolkit/mui';
+import { useWidgetApi } from '@matrix-widget-toolkit/react';
+import ImageIcon from '@mui/icons-material/Image';
+import UploadIcon from '@mui/icons-material/Upload';
+import {
+  Alert,
+  AlertTitle,
+  Box,
+  Button,
+  Card,
+  CardContent,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  Grid,
+  Input,
+  Typography,
+} from '@mui/material';
+import { unstable_useId as useId } from '@mui/utils';
+import {
+  EventDirection,
+  WidgetApiFromWidgetAction,
+  WidgetEventCapability,
+} from 'matrix-widget-api';
+import { ChangeEvent, ReactElement, useCallback, useState } from 'react';
+import { NavigationBar } from '../NavigationPage';
+import { StoreProvider } from '../store';
+import { ImageListView } from './ImageListView';
+
+import { ROOM_EVENT_UPLOADED_IMAGE, UploadedImageEvent } from '../events';
+
+/**
+ * A component that showcases how to upload image files and render them in a widget.
+ */
+export const UploadImagePage = (): ReactElement => {
+  const widgetApi = useWidgetApi();
+  const [errorDialogOpen, setErrorDialogOpen] = useState<boolean>(false);
+  const [selectedFile, setSelectedFile] = useState<File | null>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const titleId = useId();
+  const descriptionId = useId();
+
+  const handleFileChange = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      const file = event.target.files?.[0] || null;
+      if (file) {
+        setSelectedFile(file);
+      }
+    },
+    [],
+  );
+
+  const handleFileUpload = useCallback(() => {
+    const uploadImage = async () => {
+      if (selectedFile) {
+        if (!selectedFile.type.startsWith('image/')) {
+          setErrorMessage(
+            'Please select a valid image file. You can upload any image format that is supported by the browser.',
+          );
+          setErrorDialogOpen(true);
+          return;
+        }
+
+        try {
+          const getConfigMedia = (await widgetApi.getMediaConfig())[
```

Comment:
This variable name is unclear to be. I would suggest to name it something like `maxUploadSize`. Also I don't know about the conventions here, yet. If the value has a unit it may make sense to add them as well, e.g. `maxUploadSizeBytes`.

Reply:
Agree - I've refactored this and now it should be more clear.